src/main/java/com/nedap/openehr/lsp/symbolextractor/SymbolInformationExtractingListener.java

```diff
@@ -95,8 +106,18 @@ public void exitArchetype(AdlParser.ArchetypeContext ctx) {
     @Override public void enterTemplate(AdlParser.TemplateContext ctx) {
         stack.addSymbol(ctx.SYM_TEMPLATE(), ctx, "archetype", SymbolKind.Constant, StackAction.PUSH);
         stack.addSymbol(ctx.ARCHETYPE_HRID(), "archetype id", SymbolKind.Class);
-        this.archetypeId = ctx.ARCHETYPE_HRID().getText();
+        setArchetypeId(ctx.ARCHETYPE_HRID());
+    }
+
+    private void setArchetypeId(TerminalNode terminalNode) {
+        this.archetypeId = terminalNode.getText();
+        this.currentOverlayid = archetypeId;
     }
+
+    private void setOverlayId(TerminalNode terminalNode) {
+        this.currentOverlayid = archetypeId;
```

Comment:
I suspect the currentOverlayId must be get from the terminalNode (which is unused at the moment)? ArchetypeId could possibly be null

Reply:
good catch, will change!